aten/src/ATen/native/ConstantPadNd.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ATen/ATen.h"
+
+namespace at { namespace native {
+
+Tensor constant_pad_nd(const Tensor& self, IntList pad, Scalar value) {
+    auto input_sizes = self.sizes();
+    auto l_inp = self.dim();
+
+    auto l_pad = pad.size() / 2;
+    auto l_diff = l_inp - l_pad;
+    AT_CHECK(l_inp >= l_pad, "Padding length too large");
```

Comment:
Let's instead check `pad`'s length is at most double of the number of dims in `self` because `pad.size() / 2` gives wrong answer for odd numbers. Also, it would be great if this error message can be improved and says explicitly what the input is and what we expect it to be.

Reply:
the padding length must always be divisible by 2, and that's checked in the python code in `nn.functional`:
```python
    assert len(pad) % 2 == 0, 'Padding length must be divisible by 2'
```

should i just add that check here and then we can proceed using `l_pad` as an integer? there's other code that depends on the value of `pad.size() / 2` so it's not as easy as just changing this error check.

Reply:
@ezyang done!